assessmentModel/src/commonMain/kotlin/navigation/NodeState.kt

```diff
@@ -266,6 +267,7 @@ open class BranchNodeStateImpl(override val node: BranchNode, final override val
         val node = navigationPoint.node ?: throw NullPointerException("Unexpected null navigationPoint.node")
         return rootNodeController?.customNodeStateFor(node, this) ?: when (node) {
             is Question -> QuestionStateImpl(node, this)
+            is FormStep -> FormStepStateImpl(node, this)
```

Comment:
I should expect that MTB will have their own subclass or implementation of the branch node state so that they can vend their own state components, but maybe not? ¯\_(ツ)_/¯

Reply:
I don't quite have enough of a picture of how node state works to say for sure, but that sounds correct.